collector/kernel/CMakeLists.txt

```diff
@@ -159,56 +139,91 @@ set(BPF_DEPENDENCIES
   render_compile_ebpf_net
 )
 
-# bpf_preprocess.py <infile> <outfile> <debugfile>
-# Note: in this command we define _PROCESSING_BPF=1 so headers shared between
-# bpf and userland can have different code when required
+set(BPF_SKEL_HEADER "${CMAKE_BINARY_DIR}/generated/render_bpf.skel.h")
+set(BPF_OBJECT_FILE "${CMAKE_BINARY_DIR}/generated/render_bpf.bpf.o")
+set(BPFTOOL "${CMAKE_INSTALL_PREFIX}/usr/local/sbin/bpftool")
+set(CLANG clang)
+
+# Get Clang's system includes for BPF compilation
+execute_process(
+  COMMAND ${CLANG} -v -E - 
+  INPUT_FILE /dev/null
+  ERROR_VARIABLE CLANG_VERBOSE_OUTPUT
+  OUTPUT_QUIET
+)
+string(REGEX MATCH "#include <\\.\\.\\.> search starts here:(.*)End of search list\\." _ "${CLANG_VERBOSE_OUTPUT}")
+string(REGEX REPLACE "\n" ";" CLANG_INCLUDES "${CMAKE_MATCH_1}")
+set(CLANG_BPF_SYS_INCLUDES "")
+foreach(INCLUDE_PATH ${CLANG_INCLUDES})
+  string(STRIP "${INCLUDE_PATH}" INCLUDE_PATH)
+  if(NOT "${INCLUDE_PATH}" STREQUAL "")
+    list(APPEND CLANG_BPF_SYS_INCLUDES "-idirafter" "${INCLUDE_PATH}")
+  endif()
+endforeach()
+
+# Determine architecture for vmlinux.h
```

Comment:
Would this help: https://cmake.org/cmake/help/latest/variable/CMAKE_HOST_SYSTEM_PROCESSOR.html ?